src/test/java/gift/ProductManagerViewControllerTest.java

```diff
@@ -31,14 +31,14 @@ public class ProductManagerViewControllerTest {
     private ObjectMapper objectMapper;
 
     @MockitoBean
-    private ProductService productService;
+    private ProductServiceImpl productServiceImpl;
```

Comment:
이런 경우 하위 구현체에 의존할 필요 없이 상위 타입인 인터페이스로 타입을 지정해주셔도 상관없을 것 같아요. 
```suggestion
    private ProductService productService;
```